### ISSUE TITLE
Support for Misskey/Calckey API

### ISSUE THREAD

**Issue by ineffyble** [NicolasConstant/sengi#539]
Would be awesome to see this added to Sengi, since I have accounts on Misskey instances as well as Mastodon.

If this is something that is desirable for the app I might be tempted to try and do the work myself.

**Comment by NicolasConstant** [comment]
That would be a nice addition!

A PR will be highly appreciated, feel free to ask if you need insights or help to get started.

**Comment by EwanCroft** [comment]
is there any update on this? this would be brilliant!

**Comment by Silmathoron** [comment]
With the move to FireFish, most of the API should be Mastodon compatible and thus work with Sengi, as far as I understand, except for FireFish-specific things like reactions, etc.

However, there seem to be a few issues, so the compatibility is probably not 100%:

- 401 Unauthorized errors when trying to fav or reblog/boost some posts
- boost/fav icon not highlighted after boost/fav
- notifications arriving rather late compared to Mastodon servers and being duplicated between "notifications" and "mentions" sections

For the 1st error, the console shows a bunch of unrelated requests:
- first to ``/api/v1/scheduled_statuses``
- then to ``/api/v2/search?q={insert random status URL}&resolve=true`` before inserting the right URL for the last call
- all get 401 error codes

@NicolasConstant if you'd like to look into it, is there anything I can provide to help debug this? In a way it does not seem FF-specific but I don't see it on my Mastodon account...

**Comment by NicolasConstant** [comment]
Yes I've run a Firefish account in Sengi to test it quickly, there is still some incompatibilities.

That's something that must be solve on the Firefish side, when I was looking to support Pleroma properly, I opened issues on their repo to notify discrepancies with the original MastoAPI / official MastoAPI documentation. I think that's the best way since it will ensure the API implementation itself will behave exactly like any MastoAPI implementation, and therefore being 100% compatible with 3rd party clients.
